### ISSUE TITLE
bug: Minimum date in Datetime causes issues with 12-hour clock

### ISSUE THREAD

**Issue by elenche** [ionic-team/ionic-framework#25183]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [X] Nightly

### Current Behavior

Setting a minimum datetime causes noon (12pm) to be missing from the PM time picker. Instead it shows in the AM picker.
Note: I tried using the `hourCycle` property but that didn't change anything.

Current values:
`minuteValues: string = '0,15,30,45';`
`min: string = '2022-04-25T08:30:00';`
`max: string = '2022-04-25T21:30:00';`

AM picker result:
![image](https://user-images.githubusercontent.com/7271045/165079052-b7fd80df-ddd7-49d5-a8a5-d59e9bc6bd61.png)

PM picker result:
![image](https://user-images.githubusercontent.com/7271045/165079170-6b4cbe74-9c7a-4213-9de2-15e286aa9a67.png)

### Expected Behavior

The time picker should correctly display the following options:
AM picker: 8:30 - 11:45
PM picker: 12:00 - 21:30

### Steps to Reproduce

Tested both on `@ionic/angular` versions `6.1.0` and `6.1.3-dev.11650654723.120c74ee`.

Open the StackBlitz URL below and try to select 12pm.

### Code Reproduction URL

https://stackblitz.com/edit/ionic6-angular13-bptrsx?file=src%2Fapp%2Fapp.component.html

### Ionic Info

On my machine:
```
Ionic:
   Ionic CLI                     : 6.16.3 (...npm\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 6.1.3-dev.11650654723.120c74ee
   @angular-devkit/build-angular : 13.3.3
   @angular-devkit/schematics    : 13.3.3
   @angular/cli                  : 13.3.3
   @ionic/angular-toolkit        : 6.1.0

...

System:
   NodeJS : v14.16.1 (C:\Program Files\nodejs\node.exe)
   npm    : 6.14.12
   OS     : Windows 10
```

**Comment by amandaejohnston** [comment]
Thanks for the issue. I'm able to replicate this in core using the following HTML:
```
<ion-datetime
  locale="en-US"
  presentation="date-time"
  min="2022-04-25T08:30"
  max="2022-04-25T21:30"
  minute-values="0,15,30,45"
  show-default-buttons="true"
></ion-datetime>
```

**Comment by acommend** [comment]
I see the same issue with @ionic/vue 6.1.3

**Comment by pjc2007** [comment]
Yes, I have this with "@ionic/angular": "^6.1.9".

```
TS:
public min = '2022-06-30T06:00:00.000';
 public max = '2022-06-30T18:00:00.000';
 public value = '2022-06-30T10:29:06.000';

HTML:
 <ion-datetime     
      hourCycle="h12"
      [min]="min" 
      [max]="max"
      [value]="value">
  </ion-datetime>
```
So anything that makes the display 12 hours (eg the locale or hourCycle)

Also I noticed it only happens if the value is also set.
If I remove the line `[value]="value"` is is ok

**Comment by jdforsythe** [comment]
Same. Only have `ngModel` and `min` set on the end time, which is set in a very common scenario to have a minimum date equal to the start date.

```ts
public startTime = '2022-07-29T08:00:00-04';
public endTime = '2022-07-29T12:00:00-04';
```

```html
<ion-datetime
  [(ngModel)]="startTime"
  name="startTime"
  presentation="date-time"
  showDefaultButtons=true"
></ion-datetime>

<ion-datetime
  [(ngModel)]="endTime"
  name="endTime"
  presentation="date-time"
  showDefaultButtons=true"
  [min]="startTime"
></ion-datetime>
```

**Comment by liamdebeasi** [comment]
Hi there,

Here is a dev build with a proposed fix if anyone is interested in testing: `6.2.8-dev.11663343248.1fce52ab`

Install Example:

```
npm install @ionic/angular@6.2.8-dev.11663343248.1fce52ab
```

Note: You may need to test this in a local Ionic app. StackBlitz sometimes has trouble installing dev builds.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/25952, and a fix will be available in an upcoming release of Ionic Framework. Please feel free to continue testing the dev build.
